synapse/handlers/sync.py

```diff
@@ -378,7 +378,7 @@ async def ephemeral_by_room(
         sync_config = sync_result_builder.sync_config
 
         with Measure(self.clock, "ephemeral_by_room"):
-            typing_key = since_token.typing_key if since_token else "0"
+            typing_key = since_token.typing_key if since_token else 0
```

Comment:
Was `get_new_events` always expecting an int? I'm not sure I understand this change.

Reply:
Ah, sorry, I noticed this as part of some mypy fun and forgot about it. Basically `since_token.typing_key` is now an `int` so this is "just" fixing the ternary to return a consistent type